test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -33,55 +39,272 @@ using namespace dev::solidity::test;
 using namespace std;
 using namespace soltest;
 
-dev::solidity::test::ParameterList ContractABIUtils::parametersFromJson(
+namespace
+{
+
+using ParameterList = dev::solidity::test::ParameterList;
+
+size_t arraySize(string const& _arrayType)
+{
+	auto leftBrack = _arrayType.find("[");
+	auto rightBrack = _arrayType.rfind("]");
+
+	soltestAssert(
+		leftBrack != string::npos &&
+		rightBrack != string::npos &&
+		rightBrack == _arrayType.size() - 1 &&
+		leftBrack < rightBrack,
+		""
+	);
+
+	string size = _arrayType.substr(leftBrack + 1, rightBrack - leftBrack - 1);
+
+	return static_cast<size_t>(stoi(size));
+}
+
+bool isBool(string const& _type)
+{
+	return _type == "bool";
+}
+
+bool isUint(string const& _type)
+{
+	return regex_match(_type, regex{"uint\\d*"});
+}
+
+bool isInt(string const& _type)
+{
+	return regex_match(_type, regex{"int\\d*"});
+}
+
+bool isFixedBytes(string const& _type)
+{
+	return regex_match(_type, regex{"bytes\\d+"});
+}
+
+bool isBytes(string const& _type)
+{
+	return regex_match(_type, regex{"\\bbytes\\b"});
+}
+
+bool isString(string const& _type)
+{
+	return _type == "string";
+}
+
+bool isFixedBoolArray(string const& _type)
+{
+	return regex_match(_type, regex{"bool\\[\\d+\\]"});
+}
+
+bool isFixedUintArray(string const& _type)
+{
+	return regex_match(_type, regex{"uint\\d*\\[\\d+\\]"});
+}
+
+bool isFixedIntArray(string const& _type)
+{
+	return regex_match(_type, regex{"int\\d*\\[\\d+\\]"});
+}
+
+bool isFixedStringArray(string const& _type)
+{
+	return regex_match(_type, regex{"string\\[\\d+\\]"});
+}
+
+bool isTuple(string const& _type)
+{
+	return _type == "tuple";
+}
+
+bool isFixedTupleArray(string const& _type)
+{
+	return regex_match(_type, regex{"tuple\\[\\d+\\]"});
+}
+
+string functionSignatureFromABI(Json::Value const& _functionABI)
+{
+	auto inputs = _functionABI["inputs"];
+
+	string signature = {_functionABI["name"].asString() + "("};
+	size_t parameterCount = 0;
+
+	for (auto const& input: inputs)
+	{
+		parameterCount++;
+		signature += input["type"].asString();
+		if (parameterCount < inputs.size())
+			signature += ",";
+
+	}
+
+	return signature + ")";
+}
+
+}
+
+boost::optional<dev::solidity::test::ParameterList> ContractABIUtils::parametersFromJsonOutputs(
+	ErrorReporter& _errorReporter,
 	Json::Value const& _contractABI,
-	string const& _functionName
-) const
+	string const& _functionSignature
+)
 {
-	ParameterList abiParams;
+	ParameterList tailOffsetTypes;
+	ParameterList valueTypeParams;
+	ParameterList dynamicTypeParams;
+
+	ParameterList finalParams;
+
+	if (!_contractABI)
+		return boost::none;
+
 	for (auto const& function: _contractABI)
-		if (function["name"] == _functionName)
+	{
```

Comment:
You can remove the `{}`'s here

Reply:
Ah, I need them back because of the changes coming with the comment on the `return` below ;)